### PR TITLE
Add client support for projects

### LINKED ISSUE #136: feat: add support for "project" query
The Unleash API client endpoint support query for specific project, to avoid having the SDK download all feature toggles like this:

```sh
http https://app.unleash-hosted.com/demo/api/client/features?project=solar 
Authorization:56907a2fa53c1d16101d509a10b78e36190b0f918d9f122d
```

Which will only return feature flags for `projectId=solar`. 

This should be implemented by adding a new configuration to the unleash SDK to specify the "projectId" and if this option is specified it should be added to the feature-toggle query as a query param when fetching feature toggles from the Unleash API.

### PR DESCRIPTION
**Pull request by ivanklee86**
# Description

Please include a summary of the change and which issue is fixed (if any).
* Allow users to specify project when initializing UnleashClient.
* Use parameter when loading features.

Fixes #136 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Unit tests
- [x] Spec Tests
- [x] Integration tests / Manual Tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
           pylint UnleashClient
       - name: Unit tests
         run: |
-          py.test --html=pytest/results.html --junitxml=pytest/results.xml --self-contained-html --cov=UnleashClient tests/unit_tests
+          py.test --flake8 --html=pytest/results.html --junitxml=pytest/results.xml --self-contained-html --cov=UnleashClient tests/unit_tests
```
Comment (ivarconr) [inline .github/workflows/pull_request.yml:26]: why did you add flake8?

File: docs/changelog.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 ## Next version
 
+## Next version
+* (Major) Support project-based feature flag loading.
```
Comment (ivarconr) [inline docs/changelog.md:4]: Is this breaking somehow? I would argue it is a minor (feature release).

File: UnleashClient/api/features.py [hunk 3]
```diff
@@ -31,8 +33,15 @@ def get_feature_toggles(url: str,
             "UNLEASH-INSTANCEID": instance_id
         }
 
-        resp = requests.get(url + FEATURES_URL,
+        base_url = f"{url}{FEATURES_URL}"
+        base_params = {}
```
Comment (ivarconr) [inline UnleashClient/api/features.py:37]: great solve. There might come more params later :dango:

**Review by ivarconr — APPROVED** [review]

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (ilee-asapp) [reply .github/workflows/pull_request.yml:26]: `-flake8` was previously only applied to the tests via `pytest.ini`.  The argument causes some headaches with VSCode test descovery, so in the interests of being more dev-agnostic I was moving it to Make/CI until I can figure out how to fix the configs.  

(Longer term I'd like to unify/modernize linting a bit.)

File: docs/changelog.md (hunk 2 above)
Comment (ilee-asapp) [reply docs/changelog.md:4]: Oops, yus that should be a minor bump.

Comment (ivanklee86) [reply docs/changelog.md]: Oops, that should be a minor bump.

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (ivanklee86) [reply .github/workflows/pull_request.yml:26]: `-flake8` was previously only applied to the tests via `pytest.ini`. The argument causes some headaches with VSCode test discovery, so in the interests of being more dev-agnostic I was moving it to Make/CI until I can figure out how to fix the configs.

(Longer term I'd like to unify/modernize linting a bit.)
